### PR TITLE
Compatibility payum 0.15

### PR DESCRIPTION
**Pull request by zyleee**
Hi @a2xchip 

This brings compatibility for Payum 0.15.x

Cheers
Christian

### CODE REVIEW

File: Action/ConvertPaymentAction.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    {
+        RequestNotSupportedException::assertSupports($this, $request);
+
+        /** @var PaymentInterface $payment */
+        $payment = $request->getSource();
+
+        $details = ArrayObject::ensureArrayObject($payment->getDetails());
+
+        $details['VendorTxCode'] = $payment->getNumber();
+        $details['Amount'] = $payment->getTotalAmount() / 100;
```
Comment (makasim) [inline Action/ConvertPaymentAction.php]: it must not be hard coded. It should be like this: https://github.com/Payum/Payum/blob/master/src/Payum/Paypal/ExpressCheckout/Nvp/Action/ConvertPaymentAction.php#L26

Yes I know It is not related to this PR, just noticed it.

File: OnsitePaymentFactory.php [hunk 2]
```diff
@@ -4,25 +4,25 @@
...
 use Ledjin\Sagepay\Action\NotifyAction;
 use Ledjin\Sagepay\Action\StatusAction;
 use Payum\Core\Bridge\Spl\ArrayObject;
-use Payum\Core\PaymentFactory;
-use Payum\Core\PaymentFactoryInterface;
+use Payum\Core\GatewayFactoryInterface;
+use Payum\Core\GatewayFactory;
 
-class OnsitePaymentFactory implements PaymentFactoryInterface
+class OnsitePaymentFactory implements GatewayFactoryInterface
```
Comment (makasim) [inline OnsitePaymentFactory.php]: FYI I renamed all OnsiteXXX to OffsiteXXX

File: Action/ConvertPaymentAction.php (hunk 1 above)
Comment (zyleee) [inline Action/ConvertPaymentAction.php]: Makes sense, updated it. Thanks for Feedback

File: OnsitePaymentFactory.php (hunk 2 above)
Comment (zyleee) [inline OnsitePaymentFactory.php]: Updated, Thanks!

**Comment by zyleee** [comment]
Hi @a2xchip

I made some changes based on @makasim s Feedback. If you consider to merge this, please check the other pull request before that will bring compatibility with Payum-0.14.x (found an issue).

Would be great to get the other pull request merged and tagged, then I can update the README of this one.

Cheers,
Christian

**Comment by a2xchip** [comment]
@zyleee Thank you very much. It is hard to test these versions cause in all applications where i use payum i'm still on version 0.12. I'll merge it and tag after that to 0.3.0 version

**Pull request merged**